### PR TITLE
Teacher Lessons

### PR DESCRIPTION
**Pull request by dborski**
## Description
Completes User Story #12 
## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Refactor (non-breaking change which maintains existing functionality)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
## Notes
- Implements two new endpoints:
  - `api/v1/teachers/:teacher_id/lessons`
     - POST new lesson
  - `api/v1/lessons/:lesson_id`
     - GET lesson by id
## RSpec Results
- 100%

### CODE REVIEW

File: api/serializers.py [hunk 1]
```diff
@@ -8,9 +9,51 @@ class Meta:
...
+    depth = 1
+
 class LessonStudentSerializer(serializers.ModelSerializer):
   class Meta:
     model = LessonStudent
     fields = (
       "student", "lesson", "score", "mood"
-    )
\ No newline at end of file
+    )
```
Comment (melatran) [inline api/serializers.py:59]: Serializer looks great.

File: api/tests/test_models.py [hunk 2]
```diff
@@ -48,7 +50,8 @@ def test_string_representation(self):
 
 class AnswerModelTest(TestCase):
   def setUp(self):
-    lesson = Lesson.objects.create(name='lesson3', description='lesson3 Description')
+    self.teacher1 = TeacherFactory()
+    lesson = self.teacher1.lesson_set.create(name='lesson3', description='lesson3 Description')
```
Comment (melatran) [inline api/tests/test_models.py:54]: Nice additional tests for the models

File: api/tests/test_views.py [hunk 3]
```diff
@@ -36,11 +36,99 @@ def test_post(self):
...
+    response = self.client.get('/api/v1/lessons/%s/' % self.lesson1.id)
+
+    self.assertEqual(response.status_code, 200)
+    self.assertEqual(response.data['name'], self.lesson1.name)
+    self.assertIsInstance(response.data['questions'][0], dict)
+    self.assertIsInstance(response.data['questions'][1], dict)
+    self.assertIsInstance(response.data['questions'][0]['answers'][0], dict)
+    self.assertIsInstance(response.data['questions'][1]['answers'][0], dict)
+
+
```
Comment (melatran) [inline api/tests/test_views.py:126]: Tests look great

File: api/views.py [hunk 4]
```diff
@@ -14,11 +15,31 @@ class TeacherDetail(generics.RetrieveAPIView):
   queryset = Teacher.objects.all()
   serializer_class = TeacherSerializer
 
-  # def get(self, request, pk):
-  #   teacher = Teacher.objects.get(pk=pk)
 
-  #   serializer = TeacherSerializer(teacher)
-  #   return Response(serializer.data)
```
Comment (melatran) [inline api/views.py:21]: Thanks for deleting this. I wasn't sure it it was needed for any other implementations

**Review by melatran — APPROVED** [review]

File: api/serializers.py [hunk 5]
```diff
@@ -8,9 +9,51 @@ class Meta:
...
+    fields = ('id', 'question', 'reading', 'answers')
+
+class LessonSerializer(serializers.ModelSerializer):
+  questions = LessonQuestionSerializer(source='question_set', many=True, read_only=True)
+  class Meta:
+    model = Lesson
+    fields = (
+      'id', 'name', 'questions', 'teacher'
+    )
+    read_only_fields = ['questions']
```
Comment (Lithnotep) [inline api/serializers.py:51]: Nice to see all the serializers.

File: api/tests/test_views.py [hunk 6]
```diff
@@ -36,11 +36,99 @@ def test_post(self):
...
+  def test_can_post_lesson(self):
+    data = {
+      'lesson': {
+        'name': 'lessonName1',
+        'questions': [
+            {
+              'question': 'question1 description',
+              'reading': 'question1 reading',
+              'answers': [
```
Comment (Lithnotep) [inline api/tests/test_views.py:62]: great testing, lesson create got complicated.

**Review by Lithnotep — APPROVED** [review]

**Pull request merged**